### PR TITLE
Update Contributing.md

### PR DESCRIPTION
**Pull request by amaitland**
Expanding `Contributing.md` to be more specific about what is required around creating issues.

### CODE REVIEW

**Comment by amaitland** [comment]
@jornh Thoughts/comments? Am I being too harsh? :smile:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,21 +3,86 @@ Thanks for your interest in contributing to the project! Please follow these sim
...
-- Max number of columns are 132, please format your code accordingly
-- Please don't add #Regions to the code
-- When adding new files, please prepend the standard license disclaimer (just copy and paste from another source file)
-- Do your best to follow these guidelines but don't be afraid to make mistakes when trying to apply them. We are all novices in the beginning.
+### `CefSharp` vs `CEF`
+
+`CefSharp` simply takes the building blocks provided by `Cef` and attempts to provide a usable '.Net' implementation.
+The upstream [`CEF` forum](http://magpcss.org/ceforum/) is a valuable resource, if your issues seems fairly low level, then please conduct some research before posting.
+
+It maybe helpful to run the `cefclient` application and compare output with `CefSharp`. The `WinForms` and `WPF` versions use two different rending modes, `WPF` uses Offscreen Rendering (`OSR`).
```
Comment (jankurianski) [inline CONTRIBUTING.md]: rending -> rendering

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,21 +3,86 @@ Thanks for your interest in contributing to the project! Please follow these sim
...
+3. 'Ask Don't Tell' : Ask how to achieve something, **don't say it's broken** just because you haven't got it working yet!
+4. **Isolate the problem** &mdash; ideally create a reproducible example.
+5. **Include a screencast if relevant** - Is your issue about a design or front end feature or bug? The most helpful thing in the world is if we can *see* what you're talking about. Use [LICEcap](http://www.cockos.com/licecap/) to quickly and easily record a short screencast (24fps) and save it as an animated gif! Embed it directly into your GitHub issue.
+6. **Include as much info as possible!** Use the **Bug Report template** below or [click this link](https://github.com/CefSharp/CefSharp/issues/new?title=Bug%3A&body=%23%23%23%20Issue%20Summary%0A%0A%23%23%23%20Steps%20to%20Reproduce%0A%0A1.%20This%20is%20the%20first%20step%0A%0AThis%20is%20a%20bug%20because...%0A%0A%23%23%23%20Technical%20details%0A%0A*%20Ghost%20Version%3A%20master%20-%20latest%20commit%3A%20%20INSERT%20COMMIT%20REF%0A*%20Client%20OS%3A%20%0A*%20Server%20OS%3A%20%0A*%20Node%20Version%3A%20%0A*%20Browser%3A%20%0A*%20Database%3A) to start creating a bug report with the template automatically.
```
Comment (jankurianski) [inline CONTRIBUTING.md]: This link seems to be for another project.  Here is a link that encodes the Bug Report template from below:

https://github.com/CefSharp/CefSharp/issues/new?title=Bug%3A&body=**What%20is%20the%20expected%20output%3F%20What%20do%20you%20see%20instead%3F**%0A%0A**What%20steps%20will%20reproduce%20the%20problem%3F**%0A%0A**What%20version%20of%20the%20product%20are%20you%20using%3F%20On%20what%20operating%20system%3F%20x86%20or%20x64%3F**%0A%0A-%20What%20version%20are%20you%20using%3F%20Nuget%3F%20CI%20Nuget%3F%20build%20from%20a%20branch%3F%20If%20so%20which%20branch%3F%0A%0A-%20Win7%2C%20Win%208%2C%20etc%3F%0A%0A**Please%20provide%20any%20additional%20information%20below.**%0A%0A-%20A%20stack%20trace%20if%20available%2C%20any%20Exception%20information.%0A%0A-%20Does%20the%20cef%20log%20provide%20any%20relevant%20information%3F%20(By%20default%20there%20should%20be%20a%20debug.log%20file%20in%20your%20bin%20directory)%0A%0A-%20Any%20other%20background%20information%20that%27s%20relevant%3F%20Are%20you%20doing%20something%20out%20of%20the%20ordinary%3F%203rd%20party%20controls%3F

Comment (jankurianski) [inline CONTRIBUTING.md]: PS I used http://meyerweb.com/eric/tools/dencoder/ to make the link

Comment (amaitland) [inline CONTRIBUTING.md]: You got in before I had a chance to fix that :smile: 

See https://github.com/amaitland/CefSharp/commit/9022ec423cbde386e91f21bdf5c2c7d3d9628d46

**Comment by amaitland** [comment]
@jankurianski Anything you'd like to see changed? Added?

Maybe need to change the `I love cookies` part to something more technical, like `As the last line include your github username and the date in reverse e.g. amaitland 20150115`

**Comment by jankurianski** [comment]
@amaitland I don't reckon you need the verification step.  We can tell at first glance whether they've used the template or not when the issue is created.  And if they haven't use it, our first response should simply be a link to the section on creating an issue :smile: 

I think it would be also useful to include this in the bug report template:
** Does this problem also occur in the TestApp from http://cefbuilds.com? **

**Comment by amaitland** [comment]
> I don't reckon you need the verification step.

That's one of my favorite parts! Lol Removed :smile: 

> I think it would be also useful to include this in the bug report template:
> ** Does this problem also occur in the TestApp from http://cefbuilds.com? **

Added :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (jornh) [inline CONTRIBUTING.md]: Wow @jankurianski ! Sweet trick and tool :+1:

**Comment by jornh** [comment]
I could nitpick on that I'm a bit worried about that it's too long to read for those who jump to posting incomplete, not well researched, **demanding** "issues" (am I sounding too harsh now :wink:) but ....

> We can tell at first glance whether they've used the template or not when the issue is created. And if they haven't use it, our first response should simply be a link to the section on creating an issue :smile:

... should be an effective filter for us (I think we can easily spot when needed)

So to put an efficient stopgap to the recent *flurry* of noisy, time (**and** mental energy) consuming incoming issues I'll respond by the only sensible choice and merge immediately! Good work guys!

In addition I'll like to propose @jankurianski as a full contributor :+1: (but I guess that's another `question` issue we need to raise separately :wink: )

File: CONTRIBUTING.md (hunk 2 above)
Comment (jankurianski) [inline CONTRIBUTING.md]: It was actually @amaitland that found the cool trick to automatically fill in the issue, I just updated the link.  :-)

Comment (jornh) [inline CONTRIBUTING.md]: Ah, I didn't read the separate commits first before commenting - bummer! Lol!

**Comment by amaitland** [comment]
> I could nitpick on that I'm a bit worried about that it's too long to read for those who jump to posting incomplete, not well researched, demanding "issues" (am I sounding too harsh now :wink:) but ....

It probably is a little long, maybe later on we can separate it out into a few more fine grained sections?

> In addition I'll like to propose @jankurianski as a full contributor :+1: (but I guess that's another question issue we need to raise separately :wink: )

Has my vote :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (amaitland) [inline CONTRIBUTING.md]: I borrowed the idea from https://github.com/TryGhost/Ghost/blob/master/CONTRIBUTING.md#bug-reports :smile:

**Comment by jankurianski** [comment]
Cheers guys.  I am happy to join and help out wherever possible.

**Comment by jornh** [comment]
@jankurianski very much appreciated!

Ah well, lets get moving on this one ... @perlun would you be so kind to add @jankurianski to the team and/or add either @amaitland or me to the **owner** group so we can do it? :+1:

**Pull request merged**